kong/core/router.lua

```diff
@@ -530,7 +530,7 @@ function _M.new(apis)
         -- hosts/URIs/methods
         local category_idx = CATEGORIES_LOOKUP[req_category]
 
-        while category_idx <= #CATEGORIES_LOOKUP do
+        while category_idx <= #CATEGORIES do
```

Comment:
I think it'd be even better to cache this value at the top level of the module and use it instead:
```
local categories_len = #CATEGORIES
```